scripts/test/corelli/calibration/test_database.py

```diff
@@ -0,0 +1,127 @@
+
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from os import remove
+import unittest
+import numpy as np
+import pathlib
+from datetime import datetime
+
+from mantid.dataobjects import TableWorkspace
+from mantid.api import WorkspaceGroup
+from mantid.simpleapi import CreateEmptyTableWorkspace
+from corelli.calibration.database import combine_spatial_banks, init_corelli_table, save_bank_table, filename_bank_table,\
+    load_bank_table, has_valid_columns
+
+
+class TestCorelliDatabase(unittest.TestCase):
+
+    def setUp(self) -> None:
+
+        # create a mock database and tests save_bank_table and load_bank_table
+        self.database_path:str = 'data/corelli'
```

Comment:
You should use [tempfile.NamedTemporaryFile](https://docs.python.org/3.6/library/tempfile.html#tempfile.NamedTemporaryFile) for where to save the file. I think you are writing in an illegal place Something like
```python
self.database_path:str = NamedTemporaryFile(prefix='corelli', delete=False).name
```
will likely work. Don't forget to delete the file in your `cleanup` step.
filename = NamedTemporaryFile

Reply:
OK, thanks. I end up using [pathlib.TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) as a static member. It seems to be called every time I reuse it to create a file (interesting behavior). All files and directories should be cleaned at `tearDown`. Let see if it passes this time. Thanks!